src/multiplex/mod.rs

```diff
@@ -299,44 +242,70 @@ impl<T, M1, M2, B1, B2, E> Transport for T
     }
 }
 
-/*
- *
- * ===== impl NewTransport =====
- *
- */
-
-impl<F, R, T> NewTransport for F
-    where F: Fn() -> R,
-          R: IntoFuture<Item = T, Error = io::Error>,
-          T: Transport,
+impl<M1, M2, B1, B2, E> Transport for Box<Transport<In = M1, Out = M2, BodyIn = B1, BodyOut = B2, Error = E>>
+    where E: From<Error<E>> + 'static,
+          M1: 'static,
+          M2: 'static,
+          B1: 'static,
+          B2: 'static,
 {
-    type In = T::In;
-    type BodyIn = T::BodyIn;
-    type Out = T::Out;
-    type BodyOut = T::BodyOut;
-    type Error = T::Error;
-    type Item = T;
-    type Future = R::Future;
-
-    fn new_transport(&self) -> Self::Future {
-        self().into_future()
+    type In = M1;
+    type BodyIn = B1;
+    type Out = M2;
+    type BodyOut = B2;
+    type Error = E;
+
+    fn poll_read(&mut self) -> Async<()> {
+        (**self).poll_read()
+    }
+
+    fn read(&mut self) -> Poll<Frame<M2, B2, E>, io::Error> {
+        (**self).read()
+    }
+
+    fn poll_write(&mut self) -> Async<()> {
+        (**self).poll_write()
+    }
+
+    fn write(&mut self, req: Frame<M1, B1, E>) -> Poll<(), io::Error> {
+        (**self).write(req)
+    }
+
+    fn flush(&mut self) -> Poll<(), io::Error> {
+        (**self).flush()
     }
 }
 
-impl<F, R, T> NewTransport for Take<F>
-    where F: FnOnce() -> R,
-          R: IntoFuture<Item = T, Error = io::Error>,
-          T: Transport,
+impl<M1, M2, B1, B2, E> Transport for Box<Transport<In = M1, Out = M2, BodyIn = B1, BodyOut = B2, Error = E> + Send>
```

Comment:
Thinking a little more about these, I think you can omit them entirely perhaps? given:

```rust
// src/transport.rs

trait Transport { ... }
impl<T: Transport + ?Sized> Transport for Box<T> { ... }

// src/multiplex/mod.rs
trait Transport { ... }
impl<T: ::Transport<...> + ?Sized> Transport for T
```

That I think reduces everything to only one impl for `Box`